pkg/controller/jobs/leaderworkerset/leaderworkerset_reconciler.go

```diff
@@ -0,0 +1,227 @@
+	wl := &kueue.Workload{}
+	err := r.client.Get(ctx, client.ObjectKey{Name: GetWorkloadName(lws.UID, lws.Name, fmt.Sprint(index)), Namespace: lws.Namespace}, wl)
+	// Ignore if the Workload already exists or an error occurs.
+	if err == nil || client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if err := r.createPrebuildWorkload(ctx, lws, index); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) createPrebuildWorkload(ctx context.Context, lws *leaderworkersetv1.LeaderWorkerSet, index int32) error {
+	createdWorkload := r.constructWorkload(lws, index)
+	err := r.client.Create(ctx, createdWorkload)
+	if err != nil {
+		return err
+	}
+	r.record.Eventf(
+		lws, corev1.EventTypeNormal, jobframework.ReasonCreatedWorkload,
+		"Created Workload: %v", workload.Key(createdWorkload),
+	)
+	return nil
+}
+
+func (r *Reconciler) constructWorkload(lws *leaderworkersetv1.LeaderWorkerSet, index int32) *kueue.Workload {
+	wl := &kueue.Workload{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        GetWorkloadName(lws.UID, lws.Name, fmt.Sprint(index)),
+			Namespace:   lws.Namespace,
+			Finalizers:  []string{kueue.ResourceInUseFinalizerName},
+			Annotations: admissioncheck.FilterProvReqAnnotations(lws.GetAnnotations()),
+		},
+		Spec: kueue.WorkloadSpec{
+			QueueName:                   jobframework.QueueNameForObject(lws),
+			PodSets:                     r.podSets(lws),
+			MaximumExecutionTimeSeconds: jobframework.MaximumExecutionTimeSecondsForObject(lws),
+		},
+	}
+
+	wl.Annotations[podcontroller.IsGroupWorkloadAnnotationKey] = podcontroller.IsGroupWorkloadAnnotationValue
+
+	return wl
+}
+
+func (r *Reconciler) podSets(lws *leaderworkersetv1.LeaderWorkerSet) []kueue.PodSet {
+	podSets := make([]kueue.PodSet, 0, 2)
+
+	if lws.Spec.LeaderWorkerTemplate.LeaderTemplate != nil {
+		podSets = append(podSets, kueue.PodSet{
+			Name:  leaderPodSetName,
+			Count: 1,
+			Template: corev1.PodTemplateSpec{
+				Spec: *lws.Spec.LeaderWorkerTemplate.LeaderTemplate.Spec.DeepCopy(),
+			},
+			TopologyRequest: jobframework.PodSetTopologyRequest(
+				&lws.Spec.LeaderWorkerTemplate.LeaderTemplate.ObjectMeta,
+				// TODO: ???
+				ptr.To(kueuealpha.PodGroupPodIndexLabel),
+				nil,
+				nil,
+			),
+		})
+	}
+
+	defaultPodSetName := kueue.DefaultPodSetName
+	if len(podSets) > 0 {
+		defaultPodSetName = workerPodSetName
+	}
+
+	defaultPodSetCount := ptr.Deref(lws.Spec.LeaderWorkerTemplate.Size, 1)
+	if len(podSets) > 0 {
+		defaultPodSetCount--
+	}
+
+	podSets = append(podSets, kueue.PodSet{
+		Name:  defaultPodSetName,
+		Count: defaultPodSetCount,
+		Template: corev1.PodTemplateSpec{
+			Spec: *lws.Spec.LeaderWorkerTemplate.WorkerTemplate.Spec.DeepCopy(),
+		},
+		TopologyRequest: jobframework.PodSetTopologyRequest(
+			&lws.Spec.LeaderWorkerTemplate.WorkerTemplate.ObjectMeta,
+			// TODO: ???
+			ptr.To(kueuealpha.PodGroupPodIndexLabel),
```

Comment:
Same about TAS here, I don't see the need for rank-ordering for leader at all, it seems confusing.

Reply:
Yeah, that was my questions. Reverted it. Thanks!